src/environment.ts

```diff
@@ -167,82 +157,95 @@ export class SecureEnvironment {
     remap(secureValue: SecureValue, rawValue: RawValue, rawDescriptors: PropertyDescriptorMap) {
         this.createSecureRecord(secureValue, rawValue);
         for (const key in rawDescriptors) {
-            // TODO: this whole block needs cleanup and simplification
-            // avoid overriding ecma script global keys.
-            if (!ESGlobalKeys.has(key)) {
-                const secureDescriptor = getOwnPropertyDescriptor(secureValue, key);
-                if (hasOwnProperty(rawDescriptors, key)) {
-                    // avoid poisoning to only installing own properties from baseDescriptors,
-                    let rawDescriptor = rawDescriptors[key];
-                    if (hasOwnProperty(rawDescriptor, 'set')) {
-                        // setter, and probably getter branch
-                        rawDescriptor = getSecureGlobalAccessorDescriptor(this, rawDescriptor);
-                    } else if (hasOwnProperty(rawDescriptor, 'get')) {
-                        // getter only branch (e.g.: window.navigator)
-                        const { get: originalGetter } = rawDescriptor;
-                        const env = this;
-                        rawDescriptor.get = function get(): SecureValue {
-                            const value: RawValue = apply(originalGetter as () => any, env.getRawValue(this), emptyArray);
-                            return env.getSecureValue(value);
-                        };
-                    } else {
-                        // value branch
+            // TODO: this whole loop needs cleanup and simplification avoid
+            // overriding ECMA script global keys.
+            if (ESGlobalKeys.has(key) || !hasOwnProperty(rawDescriptors, key)) {
```

Comment:
what do you think about using `hasOwnProperty` for `ESGlobalKeys`, to avoid poisoning the global object's `has` method?

Reply:
We can have an uncurried `SetHas` for that. I'll review our set/map use to make sure we avoid prototype pollution.